### PR TITLE
Select testcase parameter set based upon presets

### PR DESCRIPTION
**Pull request by moosq**

### CODE REVIEW

File: project/project.go [hunk 1]
```diff
@@ -572,6 +578,47 @@ func SplitTest(name string) (string, string) {
 	return name, ""
 }
 
+// DoesTestcaseMatchPreset: check whether testcase instance shall
+// be executed dependent on the specified presets
+func DoesTestcaseMatchPreset(tc *TestConfig, presets []string) bool {
```
Comment (5nord) [inline project/project.go:583]: I would split this function into smaller ones with less complexity.
Something like bellow, if you know what I mean:

```go
func DoesTestConfigMatchPresets(tc TestConfig, presets ...string) bool {
    if tc.Except == nil && tc.Only == nil {
        return true
    }
    return !MatchPresets(tc.Exept, presets...) && MatchPresets(c.Only, presets...)
}

func MatchPresets(c *ExecuteCondition, preset ...string) bool {
    for _, p := range presets {
        for _, q := range c.Presets {
            if p == q {
                return true
            }
        }    
    }
   return false
}
```

File: project/project_test.go [hunk 2]
```diff
@@ -349,6 +349,98 @@ func TestImportTasks(t *testing.T) {
...
+          only:
+            presets:
+            - A`)
+		var p []project.TestConfig
+		p = append(p, project.AcquireExecutables(&a, []string{"bar://foo.ttcn3"}, []string{"B", "A"})...)
+		assert.Equal(t, &expected, &project.Parameters{Execute: p})
+	})
+
+}
+
```
Comment (5nord) [inline project/project_test.go:443]: I would add some corner-case-tests. For example, what happens for those configurations:
```yaml
# empty only condition
- test: tc1
  only:

# empty except condition
- test: tc2
  except:

# non-existing only condition
- test: tc3
  only:
    presets: [ "does-not-exist" ]

# conflicting conditions
- test: tc4
  only:
    presets: [ "A" ]
  except:
    presets: ["A"]

# conflicting conditions alternate order
- test: tc5
  except:
    presets: ["A"]
  only:
    presets: ["A"]

```

**Review by 5nord — APPROVED** [review]
